directions/forms/forms580.py

```diff
@@ -0,0 +1,295 @@
+import locale
+import os.path
+import sys
+from copy import deepcopy
+from io import BytesIO
+from typing import List, Union
+import simplejson as json
+from reportlab.graphics.shapes import Drawing
+from reportlab.lib import colors
+from reportlab.lib.enums import TA_CENTER, TA_JUSTIFY
+from reportlab.lib.styles import getSampleStyleSheet
+from reportlab.lib.units import mm
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, Table, TableStyle, KeepTogether, PageBreak
+from transliterate import translit
+from directions.models import Napravleniya
+from laboratory.settings import FONTS_FOLDER
+from laboratory.utils import strdate, strtime
+from podrazdeleniya.models import Podrazdeleniya
+from utils import xh
+from utils.xh import translation_number_from_decimal
+from reportlab.graphics.barcode import qr
+from reportlab.graphics.barcode import createBarcodeDrawing
+from django.utils.text import Truncator
+from django.core.paginator import Paginator
+
+
+def form_01(request_data):
+    """
+    ширина 80 мм
+    """
+
+    if sys.platform == 'win32':
+        locale.setlocale(locale.LC_ALL, 'rus_rus')
+    else:
+        locale.setlocale(locale.LC_ALL, 'ru_RU.UTF-8')
+
+    # Генерировать pdf-Лист на оплату
+    pdfmetrics.registerFont(TTFont('PTAstraSerifBold', os.path.join(FONTS_FOLDER, 'PTAstraSerif-Bold.ttf')))
+    pdfmetrics.registerFont(TTFont('PTAstraSerifReg', os.path.join(FONTS_FOLDER, 'PTAstraSerif-Regular.ttf')))
+    pdfmetrics.registerFont(TTFont('Symbola', os.path.join(FONTS_FOLDER, 'Symbola.ttf')))
+
+    buffer = BytesIO()
+    pagesize = (80 * mm, 120 * mm)
+    doc = SimpleDocTemplate(buffer, pagesize=pagesize, leftMargin=2 * mm, rightMargin=0 * mm, topMargin=1 * mm, bottomMargin=1 * mm, allowSplitting=1, title="Форма {}".format("80 мм"))
+    styleSheet = getSampleStyleSheet()
+    style = styleSheet["Normal"]
+    style.fontName = "PTAstraSerifReg"
+    style.fontSize = 8
+    style.leading = 7
+    style.spaceAfter = -1 * mm
+    style.alignment = TA_JUSTIFY
+    style.firstLineIndent = -1 * mm
+
+    styleFL = deepcopy(style)
+    styleFL.firstLineIndent = 0
+    styleFL.fontSize = 20
+
+    styleCenter = deepcopy(style)
+    styleCenter.alignment = TA_CENTER
+    styleCenter.fontSize = 10
+    styleCenter.leading = 10
+    styleCenter.spaceAfter = 0 * mm
+
+    styleCenterHospital = deepcopy(styleCenter)
+    styleCenterHospital.fontSize = 8
+
+    styleCenterTitle = deepcopy(styleCenter)
+    styleCenterTitle.fontSize = 14
+
+    styleTypeResearch = deepcopy(style)
+    styleTypeResearch.firstLineIndent = -0.5 * mm
+
+    objs: List[Union[Spacer, Paragraph, Table, KeepTogether]] = []
+
+    directions = json.loads(request_data.get("napr_id", '[]'))
+    for direction in sorted(directions):
+        dir = Napravleniya.objects.filter(pk=direction).first()
```

Comment:
Почему не
```suggestion
        dir = Napravleniya.objects.get(pk=direction)
```
Если в dir будет None, то ниже код и так и так упадет